packages/flutter_test/lib/src/instrumentation.dart

```diff
@@ -284,3 +300,42 @@ class TestGesture {
     _target.dispatchEvent(pointer.cancel(), _result);
   }
 }
+
+class _DepthFirstChildIterable extends IterableBase<Element> {
+  _DepthFirstChildIterable(this.rootElement);
+
+  Element rootElement;
+
+  @override
+  Iterator<Element> get iterator => new _DepthFirstChildIterator(rootElement);
+}
+
+class _DepthFirstChildIterator implements Iterator<Element> {
+  _DepthFirstChildIterator(Element rootElement)
+      : _stack = _reverseChildrenOf(rootElement);
+
+  Element _current;
+
+  final List<Element> _stack;
+
+  @override
+  Element get current => _current;
+
+  @override
+  bool moveNext() {
+    if (_stack.isEmpty)
+      return false;
+
+    _current = _stack.removeLast();
+    // Stack children in reverse order to traverse first branch first
+    _stack.addAll(_reverseChildrenOf(_current));
+
+    return true;
+  }
+
+  static List<Element> _reverseChildrenOf(Element element) {
+    List<Element> children = <Element>[];
+    element.visitChildren(children.add);
+    return children.reversed.toList();
```

Comment:
This should just return Iterable<Element> to avoid the extra copy.  You're just going to `addAll` them into `_stack` anyway.